File: backend/app/schemas/ai.py
from pydantic import BaseModel


class AISummaryRequest(BaseModel):
    lat: float
    lon: float
    address: str


class AISummaryOut(BaseModel):
    summary_text: str
    key_insights: list[str]
    recommendations: list[str]


class ChatMessageIn(BaseModel):
    role: str       # user / assistant
    content: str


class AIChatRequest(BaseModel):
    message: str
    lat: float
    lon: float
    address: str
    conversation_history: list[ChatMessageIn] = []


class AIChatOut(BaseModel):
    message: str
    agent: str
    sources: list[str] = []

File: frontend/src/types/ai.ts
export type MessageRole = "user" | "assistant";

export interface ChatMessage {
  id: string;
  role: MessageRole;
  content: string;
  timestamp: string;
  agent?: string; // which specialist agent responded
}

export interface AgentResponse {
  message: string;
  agent: string;
  sources?: string[];
}

export interface AISummary {
  summary_text: string;
  key_insights: string[];
  recommendations: string[];
}
